charmcraft.yaml

```
@@ -89,3 +91,5 @@ resources:
       scraping. The snap resource on Charmhub is intended to be an empty file. The charm will
       ignore an empty file resource and normally install the snap from the snap store, but a
       custom snap can also be provided here if needed.
+
+      This resource has top priority when the install hook is triggered. The snap_channel option will have no effect.
```

maybe for clarity, something like this?

```suggestion
      This resource has top priority. If the resource is present, the snap_channel option will have no effect.
```

(this is user facing documentation, so mentioning install hook triggered is implementation details they don't need to know)

---

Updated